ros2bag/ros2bag/reindexer/reindex_base.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2020 DCS Corporation, All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# DISTRIBUTION A. Approved for public release; distribution unlimited.
+# OPSEC #4584.
+#
+# Delivered to the U.S. Government with Unlimited Rights, as defined in DFARS
+# Part 252.227-7013 or 7014 (Feb 2014).
+#
+# This notice must appear in all copies of this file and its derivatives.
+
+from typing import Literal, Optional
+
+from ros2bag.api import print_error
+
+from . import reindex_sqlite
+
+
+def reindex(uri: str,
+            storage_id: str,
+            compression_fmt: Literal['', 'zstd'],
```

Comment:
I don't think this is really a `Literal` value - it'll require shotgun surgery if a new compression format is added. You should probably take an `Optional[str]` (for both fmt and mode) - the underlying logic can decide whether or not it can handle this value - overkill and brittle to handle it in the type system.